src/Npgsql/TypeHandlers/RangeHandler.cs

```diff
@@ -17,15 +22,23 @@ namespace Npgsql.TypeHandlers
     /// http://www.postgresql.org/docs/current/static/rangetypes.html
     /// </remarks>
     /// <typeparam name="TElement">the range subtype</typeparam>
-    public class RangeHandler<TElement> : NpgsqlTypeHandler<NpgsqlRange<TElement>>
+    public class RangeHandler<TElement> : NpgsqlTypeHandler<NpgsqlRange<TElement>>, IRangeHandler
     {
         /// <summary>
         /// The type handler for the element that this range type holds
         /// </summary>
         readonly NpgsqlTypeHandler _elementHandler;
 
+        public Type[] FieldTypes { get; }
+
+        public RangeHandler(NpgsqlTypeHandler elementHandler, Type[] fieldTypes)
```

Comment:
This constructor should rather be protected than public.
You're not supposed to call it with some made-up array of types that you think you could support this way.
I know you can still produce nonsense when deriving but I somehow consider deriving from a class as more "advanced" usage than just instantiating it.

Reply:
Agreed, made the change.